src/libs/import.c

```diff
@@ -294,32 +287,25 @@ static gboolean _camctl_camera_control_status_callback_gui_thread(gpointer user_
     case CAMERA_CONTROL_BUSY:
     {
       /* set all devices as inaccessible */
-      GList *list, *child;
-      list = child = gtk_container_get_children(GTK_CONTAINER(d->devices));
-      if(child)
-        do
-        {
-          if(!(GTK_IS_TOGGLE_BUTTON(child->data)
-               && gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(child->data)) == TRUE))
-            gtk_widget_set_sensitive(GTK_WIDGET(child->data), FALSE);
-        }
-        while((child = g_list_next(child)));
-
+      GList *list = gtk_container_get_children(GTK_CONTAINER(d->devices));
+      for(const GList *child = list; child; child = g_list_next(child))
+      {
+        if(!(GTK_IS_TOGGLE_BUTTON(child->data)
+             && gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(child->data)) == TRUE))
+          gtk_widget_set_sensitive(GTK_WIDGET(child->data), FALSE);
+      }
       g_list_free(list);
     }
     break;
 
     case CAMERA_CONTROL_AVAILABLE:
     {
       /* set all devices as accessible */
-      GList *list, *child;
-      list = child = gtk_container_get_children(GTK_CONTAINER(d->devices));
-      if(child)
-        do
-        {
-          gtk_widget_set_sensitive(GTK_WIDGET(child->data), TRUE);
-        }
-        while((child = g_list_next(child)));
+      GList *list = gtk_container_get_children(GTK_CONTAINER(d->devices));
```

Comment:
Looks that way, though to be sure that it's completely portable, it will need to go through a one-line adapter function.  The loop just above can also be converted to a gtk_container_foreach.

Reply:
Are you sure? I've seen several of these two-parameter function calls in the wild and assume the foreach was designed to be used that way.

Reply:
`gtk_container_foreach(GTK_CONTAINER(d->devices), (GtkCallback)gtk_widget_set_sensitive, GINT_TO_POINTER(TRUE));`

Reply:
Casting a function pointer is fraught with peril.  I've already fixed one bug in dt caused by a function-pointer cast....
- identical args except for adding/removing const is fine (but then if one type or the other changes, you need to make sure to find all the casts)
- different number of args is a definite no-no (that's what happened in the bug I fixed)
- mixing pointer/nonpointer args depends on the architecture
- one some old architectures, different pointer types had different sizes (and not just from real-mode x86 memory models)